infrastructure/private-api.yaml

```diff
@@ -0,0 +1,62 @@
+openapi: "3.0.1"
+info:
+  title: "HMRC OAuth Token Generator Private Api"
+  version: "1.0"
+
+paths:
+  /token:
+    get:
+      parameters:
+        - $ref: "#/components/parameters/token"
+      responses:
+        200:
+          description: OK
+        400:
+          description: Bad Request
+        403:
+          description: Forbidden
+        500:
+          description: Internal Server Error
+      x-amazon-apigateway-integration:
+        httpMethod: "POST"
+        passthroughBehavior: "when_no_templates"
+        type: "aws"
+        credentials:
+          Fn::Sub: ${ExecuteStateMachineRole.Arn}
+        uri:
+          Fn::Sub: arn:aws:apigateway:${AWS::Region}:states:action/StartSyncExecution
+        requestTemplates:
+          application/json:
+            Fn::Sub: |-
+              {
+                "input": "{\"tokenType\": \"$input.params('tokenType').trim()\"}",
+                "stateMachineArn": "${BearerTokenRetrievalStateMachine.Arn}"
+              }
+        responses:
+          default:
+            statusCode: 200
+            responseTemplates:
+              application/json: |
+                #set($response = $util.parseJson($input.body))
+                $response.output
+
+components:
+  parameters:
+    token:
+      name: "tokenType"
+      description: "The token to retrieve e.g. production, sandbox, stub"
```

Comment:
Hi @Caitlin-cooling,

That might not be a good idea. 

If we use an `enum` we will have to maintain it to keep up-to-date with any new data sources that we add. When accessing this through something Swagger the `tokenType` is no longer free-text which means the user is forced to use those specific values. 

![image](https://github.com/govuk-one-login/ipv-cri-otg-hmrc/assets/76599223/7c2922f7-647d-47fa-8b96-9a781fc5b996)

vs. the current free text input:

![image](https://github.com/govuk-one-login/ipv-cri-otg-hmrc/assets/76599223/338a4dab-0e03-49dd-8715-7c96b05d2bb7)

I think it's best to keep the freedom we currently have. 

Thanks

Reply:
I see what you mean. The thinking was that we need to actually be able to support a `tokenType` that is passed in, right? So would it be beneficial to stop any requests that we aren't going to be able to handle at the gateway? Plus have the added bonus of documentation of the values that are going to be successful? 

Will leave the decision up to you 👍

Reply:
I'm going to leave it as is, the APIGW doesn't actually care about the `enum` so it ignores it which means the only place it's really makes a difference is within Swagger.  I like the idea of having documentation on what values are successful but I don't think this would be best the best approach because of all the factors that might be in play. 

Thanks for the review 👍